jupyter_server_fileid/handler.py

```diff
@@ -0,0 +1,40 @@
+from jupyter_server.auth.decorator import authorized
+from jupyter_server.base.handlers import APIHandler
+from jupyter_server.extension.handler import ExtensionHandlerMixin
+from tornado import web
+from tornado.escape import json_encode
+
+from .manager import BaseFileIdManager
+
+
+class FileIDHandler(ExtensionHandlerMixin, APIHandler):
+    """Preovides a simple REST API for fetching the File ID information of a file."""
+
+    auth_resource = "contents"
+
+    @property
+    def file_id_manager(self) -> BaseFileIdManager:
+        return self.settings.get("file_id_manager")
+
+    @web.authenticated
+    @authorized
+    def get(self):
+        path: str = self.get_argument("path", None)
+        id: str = self.get_argument("id", None)
+        response = {"id": id, "path": path}
+        # If no query parameter is given, log a helpful message.
+        if not path and not id:
+            raise web.HTTPError(
+                400, log_message="No 'id' or 'path' parameters were provided in the request."
+            )
```

Comment:
Minor suggestion: Perhaps we should also raise a 400 when both the ID and path are provided.

```suggestion
        query_keys = 0
        if path:
            query_keys += 1
        if id:
            query_keys += 1
        
        # If query is invalid, raise 400.
        if not path and not id:
            raise web.HTTPError(
                400, log_message="No 'id' or 'path' parameters were provided in the request."
            )
        if query_keys != 1:
            raise web.HTTPError(
                400, log_message="Exactly one of 'id' or 'path' must be included in the query parameters."
            )
```

Reply:
Hm, do we want this return an error? 

Maybe, instead, this should validate that the id + path match what is stored in the file ID service? If there is a mismatch, then we raise an error. Otherwise, we return the data back with a 200 status code.

What do you think?

I could see this being helpful to the frontend to validate it's file ID info at any point.

Reply:
I mean, how is that any different from a client calling `GET /api/fileid?path=<path>` and verifying that the returned ID matches the one it has in memory? I don't think it really makes sense to support queries that provide both, since I haven't encountered this practice in any APIs I use.

Given that we should only support querying by one attribute or the other, we should consider splitting this endpoint into two to make the API more RESTful:

```
GET /api/fileid/id?path=<...> # gets id from path
GET /api/fileid/path?id=<...> # gets path from id
```

Let me know your thoughts.